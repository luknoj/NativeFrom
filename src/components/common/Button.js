import React from 'react';
import { TouchableOpacity, Text } from 'react-native';

const Button = (props) => {
  const { buttonStyle, textStyle } = styles;
  return (
    <TouchableOpacity onPress={props.onPress} style={buttonStyle}>
      <Text style={textStyle}> 
        {props.children}
      </Text>
    </TouchableOpacity>
  );
}

const styles = {
  buttonStyle: {
    alignSelf: 'center',
    borderRadius: 5,
    backgroundColor: '#33b5e5',
    width: 100,
    padding: 10,
    marginTop: 10,
    
  },
  textStyle: {
    alignSelf: 'center',
    fontSize: 16,
    fontWeight: '600',
    color: '#fff'
  }
}

export { Button };