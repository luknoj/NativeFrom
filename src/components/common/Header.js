import React from 'react';
import { Text, View } from 'react-native';

const Header = ({ title }) => {
  return (
    <View style={styles.headerContainer}>
      <Text style={styles.headerText}>{title}</Text>
    </View>
  );
}

const styles = {
  headerContainer: {
    height: 60,
    justifyContent: 'center',
    alignItems: 'center'
  },
  headerText: {
    fontSize: 16,
    fontWeight: '600'
  }
}

export { Header };