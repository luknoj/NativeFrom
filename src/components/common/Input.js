import React from 'react';
import { Text, View, TextInput } from 'react-native';

const Input = ({ label, value, placeholder, onChangeText, length }) => {
  const { containerStyle, labelStyle, inputStyle } = styles;

  return (
    <View style={containerStyle}>
      <Text style={labelStyle}>
        {label}
      </Text>
      <TextInput
        style={inputStyle}
        placeholder={placeholder}
        value={value}
        autoCorrect={false}
        onChangeText={onChangeText}
        maxLength={length}
      />
    </View>
  );
}

const styles = {
  containerStyle: {
    alignItems: 'center',
    flexDirection: 'row',
    height: 40,
    margin: 10,

  },
  labelStyle: {
    color: '#000',
    fontSize: 16,
    flex: 1,
    marginRight: 10,
    marginLeft: 10
  },
  inputStyle: {
    fontSize: 16,
    flex: 2.5,
    marginRight: 10,
    paddingBottom: 0,
    paddingTop: 0,
  }
}
export { Input };