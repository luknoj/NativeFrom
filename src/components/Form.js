import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { Input, Button } from './common';
import axios from 'axios';

export default class Form extends Component {
  state = {
    email: '',
    kod: '',
    imie: '',
    errorEmail: true,
    errorKod: true,
    errorImie: true,
    errorMsg: true,
    responseMsg: null,
    responseStatus: null,
  }

  validateEmail(email) {
    var emailReg = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    if(!emailReg.test(email)){
      this.setState({ errorEmail: "Wrong email", email: email });
    } else {
      this.setState({ errorEmail: false, email: email});
    }
  }

  validateName(name) {
    var imie = /^[A-Z]{1}[a-z]{2,19}$/;

    if(!imie.test(name)){
      this.setState({ errorImie: "Wrong name", imie: name });
    } else {
      this.setState({ errorImie: false, imie: name});
    }
  }

  validateZip(zip) {
    var kod = /^[0-9]{2}-[0-9]{3}$/;

    if(!kod.test(zip)){
      this.setState({ errorKod: "Wrong zip code", kod: zip });
    } else {
      this.setState({ errorKod: false, kod: zip});
    }
  }

  sendForm() { 
    if (this.state.email.length < 1 || this.state.imie.length < 1 || this.state.kod.length < 1){
      this.setState({ responseMsg: "Fields cant be empty." });
    } else {
      this.setState({ responseMsg: null });
    }
    if (!this.state.errorEmail && !this.state.errorImie && !this.state.errorKod){
      var data = {
        'zip': this.state.kod, 
        'email': this.state.email, 
        'name': this.state.imie 
      };

      this.setState({ 
        errorMsg: false,
        email: '',
        kod: '',
        imie: '',
        errorEmail: true,
        errorKod: true,
        errorImie: true,
        errorMsg: true,
      });
      
      axios.post('http://10.30.21.115:8888/api/entries/create/', data)
      .then( response => {
        console.log(response);
        if(response.status === 201){
          this.setState({ 
            responseMsg: "Your form has been successfully send.",
            responseStatus: response.status,
          })
        } else {
          this.setState({ 
            responseMsg: "Something went wrong, try again.",
            responseStatus: response.status,
          })
        }
      })
      .catch( error => {
        console.log(error);
      })
    }
  } 
  render(){
    return (
      <View>
        <Input 
          value={this.state.email} 
          placeholder="jan.kowalski@wp.pl" 
          label="Email" 
          onChangeText={ email => this.validateEmail(email) }
          length={100}
        />
        <Text style={{ alignSelf: 'center', color: 'red' }}>
          {typeof(this.state.errorEmail) === 'string' ? "Email is invalid" : null }
        </Text>

        <Input 
          value={this.state.imie} 
          placeholder="Jan" 
          label="Imię"
          onChangeText={ imie => this.validateName(imie) } 
          length={20}
        />
        <Text style={{ alignSelf: 'center', color: 'red' }}>
          {typeof(this.state.errorImie) === 'string' ? "Name is invalid" : null }
        </Text>

        <Input 
          value={this.state.kod} 
          placeholder="12-345" 
          label="Kod pocztowy"
          onChangeText={ kod => this.validateZip(kod) } 
          length={6}
        />
        <Text style={{ alignSelf: 'center', color: 'red' }}>
          {typeof(this.state.errorKod) === 'string' ? "Zip code is invalid" : null }
        </Text>

        <Text style={{ alignSelf: 'center', color: this.state.responseStatus !== 201 ? 'red' : 'green' }}>
          {this.state.responseMsg}
        </Text>

        <Button onPress={this.sendForm.bind(this)}>
          Submit
        </Button>
      </View>
    );
  }
}

