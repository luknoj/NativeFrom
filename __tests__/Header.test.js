import React from 'react';
import { Header } from '../src/components/common/';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const header = renderer.create(<Header title="test" />).toJSON();
  expect(header).toMatchSnapshot();
});