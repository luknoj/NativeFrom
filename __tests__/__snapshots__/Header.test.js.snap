// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    test
  </Text>
</View>
`;
